deploy-board/deploy_board/webapp/env_views.py

```diff
@@ -859,35 +860,37 @@ def post_add_stage(request, name):
     if from_stage and from_stage not in stages:
         raise Exception("Can not clone from non-existing stage!")
 
+    external_id = None
     if from_stage:
         try:
             external_id = environs_helper.create_identifier_for_new_stage(request, name, stage)
             common.clone_from_stage_name(request, name, stage, name, from_stage, description, external_id)
         except TeletraanException as detail:
+            message = 'Failed to create stage {}/{}: {}'.format(name, stage, detail)
+            log.error(message)
+            messages.add_message(request, messages.ERROR, message)
             if external_id:
                 try:
                     environs_helper.delete_nimbus_identifier(request, external_id)
-                except:
+                except TeletraanException as detail:
```

Comment:
Seems this could slightly change the exception handling behavior. Is it needed?

Reply:
I added it to ensure continuity between this line and line 890 and the exception that is going to come from the nimbusclient.py that handles the nimbus Response is going to be a `TeletraanException`, so I think it might be necessary to handle it and output the detail.